### PR TITLE
Detector-Competition-Feat: Add Metabase Session Secret Detector

### PR DESCRIPTION
**Pull request by fumblehool**
### Description:
PR adds detector for https://www.metabase.com/

As per the authentication flow, user will follow https://www.metabase.com/learn/administration/metabase-api#getting-started-with-the-metabase-api to generate Session Token.

After this, this token is passed in Header as `X-Metabase-Session`.

### Verification:

Screenshot 2023-10-19 at 5 06 52 PM

### CODE REVIEW

**Comment by zricethezav** [comment]
hi @fumblehool, thanks for the contributions and participation in our Hacktoberfest detector competition. Would you mind completing the documentation steps as listed on the detector competition page? https://github.com/trufflesecurity/hacktoberfest2023/blob/main/README.md#1-documentation 

We need a screenshot of passing tests

File: pkg/detectors/metabase/metabase.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"metabase"}) + `\b([a-zA-Z0-9-]{36})\b`)
+
+	baseURL = regexp.MustCompile(detectors.PrefixRegex([]string{"metabase"}) + `\b(https?:\/\/[-a-zA-Z0-9@:%._\+~#=]{1,256})\b`)
```
Comment (zricethezav) [inline pkg/detectors/metabase/metabase.go:25]: @fumblehool 
> {1,256}

I'm concerned about the lower bound here, any way we can bump that up? Seems unlikely we would get anything less than 7 here

**Comment by fumblehool** [comment]
@zricethezav Updated the code, please review and merge 😃

**Review by ahrav — APPROVED** [review]
Nice! LGTM 👍

**Pull request merged**